File: infoscreens-2/frontend/cms/src/app/models/video/apiVideo_Publish.ts
import IStringDictionary from '@interfaces/IStringDictionary';
import { VideoBackground } from '../../../../../common';

export class apiVideo_Publish {
    id: number | null;
    title: IStringDictionary<string>;
    url: string;
    duration: number;
    background: VideoBackground | null;
    isVisible: boolean;
    publicationDate: string;
    expirationDate: string;
    assignedToInfoscreenIds: number[];
    isForInfoscreens: boolean;
    isForApp: boolean;
    description: string;
    categories: string[];
    
    constructor(
        id: number | null,
        title: IStringDictionary<string>,
        url: string,
        duration: number,
        background: VideoBackground | null,
        isVisible: boolean,
        publicationDate: string,
        expirationDate: string | null,
        assignedToInfoscreenIds: number[],
        isForInfoscreens: boolean,
        isForApp: boolean,
        description: string,
        categories: string[]
    ) {
        this.id = id;
        this.title = title;
        this.url = url;
        this.duration = duration;
        this.background = background;
        this.isVisible = isVisible;
        this.publicationDate = publicationDate;
        this.expirationDate = expirationDate;
        this.assignedToInfoscreenIds = assignedToInfoscreenIds;
        this.isForInfoscreens = isForInfoscreens;
        this.isForApp = isForApp;
        this.description = description;
        this.categories = categories;
    }
}

File: infoscreens-2/backend/Infoscreens.common/Models/API/CMS/Video/apiVideo_Publish.cs
using Infoscreens.Common.Enumerations;
using Infoscreens.Common.Helpers;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace Infoscreens.Common.Models.API.CMS
{
    public class apiVideo_Publish
    {
        [JsonProperty(Required = Required.AllowNull)]
        public int? Id { get; set; }

        [JsonProperty(Required = Required.Always)]
        public Dictionary<string, string> Title { get; set; }

        [JsonProperty(Required = Required.Always)]
        public string Url { get; set; }

        [JsonProperty(Required = Required.Always)]
        public int Duration{ get; set; }

        [JsonProperty(Required = Required.AllowNull)]
        public eVideoBackground? Background { get; set; }

        [JsonProperty(Required = Required.Always)]
        public bool IsVisible { get; set; }

        [JsonProperty(Required = Required.Always)]
        public DateTimeOffset PublicationDate { get; set; }

        [JsonProperty(Required = Required.AllowNull)]
        public DateTimeOffset? ExpirationDate { get; set; }

        [JsonProperty(Required = Required.AllowNull)]
        public List<int> AssignedToInfoscreenIds { get; set; }

        [JsonProperty(Required = Required.Always)]
        public bool IsForInfoscreens { get; set; }

        [JsonProperty(Required = Required.Always)]
        public bool IsForApp { get; set; }

        [JsonProperty(Required = Required.AllowNull)]
        public string Description { get; set; }

        [JsonProperty(Required = Required.AllowNull)]
        public List<string> Categories { get; set; }


        // Methods

        internal bool CheckConsistancy()
        {
            var isUrlValid = UrlHelper.IsYoutubeWebsiteUrl(Url);

            return isUrlValid;
        }
    }
}
